### PR TITLE
Perf improvements

### PR DESCRIPTION
**Pull request by cdmihai**
Found these two by looking at low hanging profiler fruits. RPS results claim that they increase a scenario by 19%, and in local tests they improve the evaluation times of a 28 project solution by about 60ms.

### CODE REVIEW

**Comment by cdmihai** [comment]
@dotnet-bot test Windows_NT Build for CoreCLR please

**Comment by cdmihai** [comment]
@dotnet-bot test Windows_NT Build for Full please

**Comment by cdmihai** [comment]
@dotnet-bot test Windows_NT Build for Full please

**Comment by cdmihai** [comment]
@dotnet-bot test Windows_NT Build for CoreCLR please

File: src/Build/Evaluation/Evaluator.cs [hunk 1]
```diff
@@ -243,6 +244,11 @@ internal class Evaluator<P, I, M, D>
 #endif
 
         /// <summary>
+        /// The search paths are machine specific and should not change during builds
+        /// </summary>
+        private static readonly EngineFileUtilities.IOCache _fallbackSearchPathsCache = new EngineFileUtilities.IOCache();
```
Comment (rainersigwald) [inline src/Build/Evaluation/Evaluator.cs]: Talked about this with @AndyGerlicher on Friday. We're a bit concerned about this cache having process lifetime. VS is long-lived and someone could conceivably install things while it's opened.

Unfortunately, I don't know how to reduce this concern.

Comment (rainersigwald) [inline src/Build/Evaluation/Evaluator.cs]: We discussed this today. The directories that are cached here are: the VS instance's MSBuild directory, and the program files MSBuild directory (just those directories, not their subdirectories). Both should require VS being closed to be created, so the process-lifetime cache should be fine.

**Pull request merged**